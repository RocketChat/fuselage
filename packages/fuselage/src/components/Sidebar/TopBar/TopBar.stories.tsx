import { Title, Description, Primary, Stories } from '@storybook/addon-docs';
import type { ComponentStory, ComponentMeta } from '@storybook/react';
import React from 'react';

import {
  TopBar,
  TopBarAction,
  TopBarActions,
  TopBarAvatar,
  TopBarTitle,
  TopBarToolBox,
} from '.';
import { SidebarSection } from '..';
import { Avatar } from '../..';

export default {
  title: 'Sidebar/TopBar',
  component: TopBar,
  parameters: {
    docs: {
      description: {
        component: 'Sidebar TopBar and ToolBox.',
      },
      page: () => (
        <>
          <Title />
          <Description />
          <Primary />
          <Stories title={''} />
        </>
      ),
    },
  },
} as ComponentMeta<typeof TopBar>;

export const Default: ComponentStory<typeof TopBar> = () => (
  <>
    <SidebarSection>
      <Avatar
        size={TopBarAvatar.size}
        alt=''
        url={
          'data:image/jpeg;base64,/9j/4AAQSkZJRgABAQAAAQABAAD/2wBDAAgGBgcGBQgHBwcJCQgKDBQNDAsLDBkSEw8UHRofHh0aHBwgJC4nICIsIxwcKDcpLDAxNDQ0Hyc5PTgyPC4zNDL/2wBDAQkJCQwLDBgNDRgyIRwhMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjL/wAARCAAoACgDASIAAhEBAxEB/8QAGwAAAgIDAQAAAAAAAAAAAAAAAAcEBgIDBQj/xAAuEAACAQQAAwcEAQUAAAAAAAABAgMABAUREiExBhMUIkFRYQcWcYGhFTJSgpH/xAAYAQADAQEAAAAAAAAAAAAAAAACAwQBAP/EAB4RAAIBBQEBAQAAAAAAAAAAAAABAgMREiExE0HR/9oADAMBAAIRAxEAPwBuXuIkhBuMe5ib/AHQP49q4L3mLitryTLTSpOiHQI5k/HzXa/qbFOEudVTu1dumWvcTaNCZYZ7vU6g6LxqjOU/24dfs1Ouh9FnkMpd3Reeyx83hAxZZEhkdV9/MBrX71WGPvJcqrJBGveKATtuXXqNU0pu02bTHXD/AGvJAluyxxRd6F4x00o+NdKoVrjbzJdvVe1t5cVLc2ck8qjnohgpPtz2v7G6JtPQ2VJwjlcw+37mchpnK6GtIuv5NFWeTsLNPvxWTvpfjvOEfwKKzEVkSct2vscS/BIzSN0YRkeX81UpPqO8masJETu7OOccY4dswYFQeftv096XV5knuJGdm2T1+agvMXj8jEaHX905QihabvcbuS7X566mLWLwSY8PuRnk/u4eZ0deTl71Ef6hY+0yM88TzeNZY4luYwpVYyduOfrvhPTnr0pXSX9y5mCsyJMdyxxvwq599em+taItqCSNc90ChvZRUruUcT0JiO18Elpk7t8v41LWzacxkBSuvjQ/FFJayjDWrCTepAQ2vUH0oo/Jk3ovpwJJeVCP5CN+lFFaaMqy+nAyuChvrTI2kN9JAsi2ZOy4IBHMnkSCP+iqBexSWdxLazoUljJVlPUH2oorkV10pRc7b1zXb/hZOzuJvM86QWEXeELxOzHSIPcmiiiunVlF2RNTpRkrs//Z'
        }
      />
      <TopBarActions>
        <TopBarAction icon='home' title='home' />
        <TopBarAction icon='magnifier' title='search' />
        <TopBarAction icon='globe' title='discover' />
        <TopBarAction icon='sort' title='sort' />
        <TopBarAction icon='edit-rounded' title='edit' />
      </TopBarActions>
    </SidebarSection>
    <TopBarToolBox>
      <TopBarTitle>Title</TopBarTitle>
      <TopBarActions>
        <TopBarAction success icon='phone' title='call' />
        <TopBarAction icon='message-disabled' title='message' />
      </TopBarActions>
    </TopBarToolBox>
    <TopBarToolBox>
      <TopBarTitle>
        Long Title Long Title Long Title Long Title Long Title Long Title Long
        Title Long Title Long Title Long Title Long Title Long Title Long Title
        Long Title Long Title Long Title Long Title Long Title Long Title Long
        Title Long Title
      </TopBarTitle>
      <TopBarActions>
        <TopBarAction icon='dialpad' title='dialpad' />
        <TopBarAction success icon='phone' title='call' />
        <TopBarAction danger icon='phone-off' title='end' />
        <TopBarAction icon='message' title='message' />
        <TopBarAction icon='contact' title='contact' />
      </TopBarActions>
    </TopBarToolBox>
  </>
);
