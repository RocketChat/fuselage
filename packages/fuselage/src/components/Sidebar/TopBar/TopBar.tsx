import type { ReactNode } from 'react';
import React from 'react';

type TopBarProps = {
  children?: ReactNode;
  className?: string;
};

export const TopBar = ({ className, ...props }: TopBarProps) => (
  <div
    className={['rc-box rc-box--full rcx-sidebar-topbar', className]
      .filter(Boolean)
      .join(' ')}
    {...props}
  />
);
