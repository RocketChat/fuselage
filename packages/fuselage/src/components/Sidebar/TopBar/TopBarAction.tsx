import type { ComponentProps, Ref } from 'react';
import React, { forwardRef } from 'react';

import { SidebarAction } from '../SidebarActions';

type TopBarActionProps = ComponentProps<typeof SidebarAction>;

export const TopBarAction = forwardRef(function TopBarAction(
  props: TopBarActionProps,
  ref: Ref<HTMLElement>
) {
  return <SidebarAction ref={ref} {...props} />;
});
