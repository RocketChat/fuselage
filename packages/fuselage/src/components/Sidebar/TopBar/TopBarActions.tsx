import type { ComponentProps, Ref } from 'react';
import React, { forwardRef } from 'react';

import { SidebarActions } from '../SidebarActions';

type TopBarActionsProps = ComponentProps<typeof SidebarActions>;

export const TopBarActions = forwardRef(function TopBarActions(
  props: TopBarActionsProps,
  ref: Ref<HTMLDivElement>
) {
  return <SidebarActions ref={ref} {...props} />;
});
