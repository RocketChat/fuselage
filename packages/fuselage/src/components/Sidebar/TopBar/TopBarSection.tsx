import React from 'react';

import { SidebarDivider } from '../SidebarDivider';
import { TopBar } from './TopBar';
import { TopBarWrapper } from './TopBarWrapper';

type TopBarSectionProps = {
  children?: React.ReactNode;
  className?: string;
};

export const TopBarSection = ({
  className,
  children,
  ...props
}: TopBarSectionProps) => (
  <TopBar
    className={['rcx-sidebar-topbar--section', className]
      .filter(Boolean)
      .join(' ')}
    {...props}
  >
    <TopBarWrapper children={children} />
    <SidebarDivider />
  </TopBar>
);
