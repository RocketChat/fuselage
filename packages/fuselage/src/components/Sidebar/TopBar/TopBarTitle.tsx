import type { ReactNode } from 'react';
import React from 'react';

import Box from '../../Box';

type TopBarTitleProps = {
  children?: ReactNode;
};

export const TopBarTitle = (props: TopBarTitleProps) => (
  <Box className='rcx-sidebar-topbar__title' withTruncatedText {...props} />
);
