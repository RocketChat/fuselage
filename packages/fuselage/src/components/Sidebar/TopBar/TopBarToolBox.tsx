import type { ReactNode } from 'react';
import React from 'react';

import { SidebarDivider } from '../SidebarDivider';
import { TopBar } from './TopBar';
import { TopBarWrapper } from './TopBarWrapper';

type TopBarToolBoxProps = {
  children?: ReactNode;
  className?: string;
};

export const TopBarToolBox = ({
  children,
  className,
  ...props
}: TopBarToolBoxProps) => (
  <TopBar
    className={['rcx-sidebar-topbar--toolbox', className]
      .filter(Boolean)
      .join(' ')}
    {...props}
  >
    <TopBarWrapper children={children} />
    <SidebarDivider />
  </TopBar>
);
