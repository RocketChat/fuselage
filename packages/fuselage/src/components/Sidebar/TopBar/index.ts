import { SidebarDivider as TopBarDivider } from '../SidebarDivider';
import { TopBar } from './TopBar';
import { TopBarAction } from './TopBarAction';
import { TopBarActions } from './TopBarActions';
import { TopBarSection } from './TopBarSection';
import { TopBarTitle } from './TopBarTitle';
import { TopBarToolBox } from './TopBarToolBox';
import { TopBarWrapper } from './TopBarWrapper';

const Avatar: { size: 'x24' } = { size: 'x24' };

export const TopBarAvatar = Avatar;

export default Object.assign(TopBar, {
  /** @deprecated use named import instead */
  Section: TopBarSection,
  /** @deprecated use named import instead */
  ToolBox: TopBarToolBox,
  /** @deprecated use named import instead */
  Wrapper: TopBarWrapper,
  /** @deprecated use named import instead */
  Avatar: TopBarAvatar,
  /** @deprecated use named import instead */
  Actions: TopBarActions,
  /** @deprecated use named import instead */
  Action: TopBarAction,
  /** @deprecated use named import instead */
  Divider: TopBarDivider,
  /** @deprecated use named import instead */
  Title: TopBarTitle,
});

export * from './TopBar';
export * from './TopBarAction';
export * from './TopBarActions';
export * from './TopBarSection';
export * from './TopBarTitle';
export * from './TopBarToolBox';
export * from './TopBarWrapper';
