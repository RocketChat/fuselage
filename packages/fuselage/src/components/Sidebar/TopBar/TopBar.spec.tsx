import { composeStories } from '@storybook/testing-react';
import { render } from '@testing-library/react';
import { axe } from 'jest-axe';
import React from 'react';

import * as stories from './TopBar.stories';

const testCases = Object.values(composeStories(stories)).map((Story) => [
  Story.storyName || 'Story',
  Story,
]);

describe('[TopBar Rendering]', () => {
  test.each(testCases)(
    `renders %s without crashing`,
    async (_storyname, Story) => {
      const tree = render(<Story />);
      expect(tree.baseElement).toMatchSnapshot();
    }
  );

  test.each(testCases)(
    '%s should have no a11y violations',
    async (_storyname, Story) => {
      const { container } = render(<Story />);

      const results = await axe(container);
      expect(results).toHaveNoViolations();
    }
  );
});
