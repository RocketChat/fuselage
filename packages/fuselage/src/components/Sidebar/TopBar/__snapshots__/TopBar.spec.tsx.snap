// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[TopBar Rendering] renders Default without crashing 1`] = `
<body>
  <div>
    <div
      class="rcx-box rcx-box--full rcx-sidebar-section"
    >
      <figure
        class="rcx-box rcx-box--full rcx-avatar rcx-avatar--x24"
      >
        <img
          alt=""
          class="rcx-avatar__element rcx-avatar__element--x24"
          src="data:image/jpeg;base64,/9j/4AAQSkZJRgABAQAAAQABAAD/2wBDAAgGBgcGBQgHBwcJCQgKDBQNDAsLDBkSEw8UHRofHh0aHBwgJC4nICIsIxwcKDcpLDAxNDQ0Hyc5PTgyPC4zNDL/2wBDAQkJCQwLDBgNDRgyIRwhMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjL/wAARCAAoACgDASIAAhEBAxEB/8QAGwAAAgIDAQAAAAAAAAAAAAAAAAcEBgIDBQj/xAAuEAACAQQAAwcEAQUAAAAAAAABAgMABAUREiExBhMUIkFRYQcWcYGhFTJSgpH/xAAYAQADAQEAAAAAAAAAAAAAAAACAwQBAP/EAB4RAAIBBQEBAQAAAAAAAAAAAAABAgMREiExE0HR/9oADAMBAAIRAxEAPwBuXuIkhBuMe5ib/AHQP49q4L3mLitryTLTSpOiHQI5k/HzXa/qbFOEudVTu1dumWvcTaNCZYZ7vU6g6LxqjOU/24dfs1Ouh9FnkMpd3Reeyx83hAxZZEhkdV9/MBrX71WGPvJcqrJBGveKATtuXXqNU0pu02bTHXD/AGvJAluyxxRd6F4x00o+NdKoVrjbzJdvVe1t5cVLc2ck8qjnohgpPtz2v7G6JtPQ2VJwjlcw+37mchpnK6GtIuv5NFWeTsLNPvxWTvpfjvOEfwKKzEVkSct2vscS/BIzSN0YRkeX81UpPqO8masJETu7OOccY4dswYFQeftv096XV5knuJGdm2T1+agvMXj8jEaHX905QihabvcbuS7X566mLWLwSY8PuRnk/u4eZ0deTl71Ef6hY+0yM88TzeNZY4luYwpVYyduOfrvhPTnr0pXSX9y5mCsyJMdyxxvwq599em+taItqCSNc90ChvZRUruUcT0JiO18Elpk7t8v41LWzacxkBSuvjQ/FFJayjDWrCTepAQ2vUH0oo/Jk3ovpwJJeVCP5CN+lFFaaMqy+nAyuChvrTI2kN9JAsi2ZOy4IBHMnkSCP+iqBexSWdxLazoUljJVlPUH2oorkV10pRc7b1zXb/hZOzuJvM86QWEXeELxOzHSIPcmiiiunVlF2RNTpRkrs//Z"
        />
      </figure>
      <div
        class="rcx-button-group rcx-button-group--align-start"
        role="group"
      >
        <button
          class="rcx-box rcx-box--full rcx-button--small-square rcx-button--square rcx-button--icon rcx-button  rcx-button-group__item"
          title="home"
          type="button"
        >
          <i
            aria-hidden="true"
            class="rcx-box rcx-box--full rcx-icon--name-home rcx-icon rcx-css-4pvxx3"
          >
            
          </i>
        </button>
        <button
          class="rcx-box rcx-box--full rcx-button--small-square rcx-button--square rcx-button--icon rcx-button  rcx-button-group__item"
          title="search"
          type="button"
        >
          <i
            aria-hidden="true"
            class="rcx-box rcx-box--full rcx-icon--name-magnifier rcx-icon rcx-css-4pvxx3"
          >
            
          </i>
        </button>
        <button
          class="rcx-box rcx-box--full rcx-button--small-square rcx-button--square rcx-button--icon rcx-button  rcx-button-group__item"
          title="discover"
          type="button"
        >
          <i
            aria-hidden="true"
            class="rcx-box rcx-box--full rcx-icon--name-globe rcx-icon rcx-css-4pvxx3"
          >
            
          </i>
        </button>
        <button
          class="rcx-box rcx-box--full rcx-button--small-square rcx-button--square rcx-button--icon rcx-button  rcx-button-group__item"
          title="sort"
          type="button"
        >
          <i
            aria-hidden="true"
            class="rcx-box rcx-box--full rcx-icon--name-sort rcx-icon rcx-css-4pvxx3"
          >
            
          </i>
        </button>
        <button
          class="rcx-box rcx-box--full rcx-button--small-square rcx-button--square rcx-button--icon rcx-button  rcx-button-group__item"
          title="edit"
          type="button"
        >
          <i
            aria-hidden="true"
            class="rcx-box rcx-box--full rcx-icon--name-edit-rounded rcx-icon rcx-css-4pvxx3"
          >
            
          </i>
        </button>
      </div>
    </div>
    <div
      class="rc-box rc-box--full rcx-sidebar-topbar rcx-sidebar-topbar--toolbox"
    >
      <div
        class="rc-box rc-box--full rcx-sidebar-topbar__wrapper"
      >
        <div
          class="rcx-box rcx-box--full  rcx-sidebar-topbar__title rcx-css-1te28na"
        >
          Title
        </div>
        <div
          class="rcx-button-group rcx-button-group--align-start"
          role="group"
        >
          <button
            class="rcx-box rcx-box--full rcx-button--small-square rcx-button--icon-success rcx-button--square rcx-button--icon rcx-button  rcx-button-group__item"
            title="call"
            type="button"
          >
            <i
              aria-hidden="true"
              class="rcx-box rcx-box--full rcx-icon--name-phone rcx-icon rcx-css-4pvxx3"
            >
              
            </i>
          </button>
          <button
            class="rcx-box rcx-box--full rcx-button--small-square rcx-button--square rcx-button--icon rcx-button  rcx-button-group__item"
            title="message"
            type="button"
          >
            <i
              aria-hidden="true"
              class="rcx-box rcx-box--full rcx-icon--name-message-disabled rcx-icon rcx-css-4pvxx3"
            >
              
            </i>
          </button>
        </div>
      </div>
      <hr
        class="rcx-box rcx-box--full rcx-sidebar--divider rcx-divider rcx-css-oz39hi"
      />
    </div>
    <div
      class="rc-box rc-box--full rcx-sidebar-topbar rcx-sidebar-topbar--toolbox"
    >
      <div
        class="rc-box rc-box--full rcx-sidebar-topbar__wrapper"
      >
        <div
          class="rcx-box rcx-box--full  rcx-sidebar-topbar__title rcx-css-1te28na"
        >
          Long Title Long Title Long Title Long Title Long Title Long Title Long Title Long Title Long Title Long Title Long Title Long Title Long Title Long Title Long Title Long Title Long Title Long Title Long Title Long Title Long Title
        </div>
        <div
          class="rcx-button-group rcx-button-group--align-start"
          role="group"
        >
          <button
            class="rcx-box rcx-box--full rcx-button--small-square rcx-button--square rcx-button--icon rcx-button  rcx-button-group__item"
            title="dialpad"
            type="button"
          >
            <i
              aria-hidden="true"
              class="rcx-box rcx-box--full rcx-icon--name-dialpad rcx-icon rcx-css-4pvxx3"
            >
              
            </i>
          </button>
          <button
            class="rcx-box rcx-box--full rcx-button--small-square rcx-button--icon-success rcx-button--square rcx-button--icon rcx-button  rcx-button-group__item"
            title="call"
            type="button"
          >
            <i
              aria-hidden="true"
              class="rcx-box rcx-box--full rcx-icon--name-phone rcx-icon rcx-css-4pvxx3"
            >
              
            </i>
          </button>
          <button
            class="rcx-box rcx-box--full rcx-button--small-square rcx-button--icon-danger rcx-button--square rcx-button--icon rcx-button  rcx-button-group__item"
            title="end"
            type="button"
          >
            <i
              aria-hidden="true"
              class="rcx-box rcx-box--full rcx-icon--name-phone-off rcx-icon rcx-css-4pvxx3"
            >
              
            </i>
          </button>
          <button
            class="rcx-box rcx-box--full rcx-button--small-square rcx-button--square rcx-button--icon rcx-button  rcx-button-group__item"
            title="message"
            type="button"
          >
            <i
              aria-hidden="true"
              class="rcx-box rcx-box--full rcx-icon--name-message rcx-icon rcx-css-4pvxx3"
            >
              
            </i>
          </button>
          <button
            class="rcx-box rcx-box--full rcx-button--small-square rcx-button--square rcx-button--icon rcx-button  rcx-button-group__item"
            title="contact"
            type="button"
          >
            <i
              aria-hidden="true"
              class="rcx-box rcx-box--full rcx-icon--name-contact rcx-icon rcx-css-4pvxx3"
            >
              
            </i>
          </button>
        </div>
      </div>
      <hr
        class="rcx-box rcx-box--full rcx-sidebar--divider rcx-divider rcx-css-oz39hi"
      />
    </div>
  </div>
</body>
`;
